Added support for docker secrets

DennisGaida opened the pull request:
Right now the bouncer API key needs to be hardcoded in the compose file and/or in the external `.env`. The current best practice is to leverage [docker secrets](https://docs.docker.com/engine/swarm/secrets/#defining-and-using-secrets-in-compose-files) that are not only supported in swarm mode, but also from docker compose.

I have added a basic implementation within the `RequiredEnv` function, that allows the use of `<environment_variable_name>+_FILE` instead of `<environment_variable_name>`. So `CROWDSEC_BOUNCER_API_KEY` becomes `CROWDSEC_BOUNCER_API_KEY_FILE`. If said `[...]_FILE` variable exists the value is read from the file and returned from the `RequiredEnv` function.

I'd greatly appreciate some testing since logging from `config.go` seems kind of impossible? Even the existing `log.Fatalf()` entries I never get to see in the docker logs (though the container doesn't start).

Example docker-compose:
```yml
version: "3.8"

secrets:
  crowdsec_bouncer_api_key:
    file: /secretsPath/crowdsec_bouncer_api_key

services:
  bouncer:
    build: .
    image: fbonalair/traefik-crowdsec-bouncer
    container_name: bouncer
    environment:
      CROWDSEC_BOUNCER_API_KEY_FILE: /run/secrets/crowdsec_bouncer_api_key
      CROWDSEC_AGENT_HOST: crowdsec:8123
    secrets:
      - crowdsec_bouncer_api_key
```

DennisGaida committed:
added support for reading environment variables from files

fbonalair commented on config/config.go:
```diff
@@ -18,12 +18,19 @@ func OptionalEnv(varName string, optional string) string {
 
 /*
-	Check for an environment variable value, exit program if not found
+	Check for an environment variable value or the equivalent docker secret, exit program if not found
```
I would prefer not to mention and link the service to a specific technology. For example, the container and your idea are also compatible with Kubernetes.
Could you change it to something like "value or a file".

fbonalair commented on config/config.go:
```diff
@@ -18,12 +18,19 @@ func OptionalEnv(varName string, optional string) string {
 func RequiredEnv(varName string) string {
 	envVar := os.Getenv(varName)
-	if envVar == "" {
+	envVarFileName := os.Getenv(varName + "_FILE")
```
Instead of adding a "_FILE" suffix, then you have a value, could you detect if the environment value is a valid file path ? For example with the function [os.Stat(path)](https://pkg.go.dev/os#File.Stat).
Like that, you have one less nest in the if statement, so it's more streamlined while still being able to get different logs for direct value vs file value.

fbonalair commented on config/config.go:
```diff
@@ -18,12 +18,19 @@ func OptionalEnv(varName string, optional string) string {
+	} else if envVar == "" && envVarFileName != "" {
+		envVarFromFile, err := os.ReadFile(envVarFileName)
+		if err != nil {
+			log.Fatalf("Could not read env var from file %s. Exiting", envVarFileName)
```
First, nice error message ! That would help the user.
Second, to go further, also send the "root" error back to the user. That would make debugging easier : was it a miss-configuration and file is not present? Or app can't read the file because of permission ? etc etc

fbonalair reviewed (changes requested):
Thank you for your contribution, that's a great idea.
My design of the app is not the best, so testing is not as easy as I would like ...
I have put some feedback to your change. Once they are fine I will merge it and add comments in the README.

DennisGaida committed:
Removed reference to docker

DennisGaida committed:
output the error message when readfile fails

note: ReadFile doesn't guarantee any specific behavior except err==nil when successful.

johnweland commented:
Looks like I wasn't the only one with this idea. The bouncer is a great piece of code, hopefully it can get updated with this feature
